src/controllers/confirmation.controller.ts

```diff
@@ -25,20 +26,28 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
   if (!email) {
     return next(createMissingError("User Email"));
   }
-  const token = req.chSession.accessToken();
-  const request = sessionService.getRequest(req.chSession);
-  if (token && request) {
-    try {
-      await apiClient.callProcessorApi(companyNum, token, request.extension_request_id);
-    } catch (e) {
-      logger.error("Error processing application " + JSON.stringify(e));
-      return next(e);
-    }
-    if (activeFeature(process.env.FEATURE_REQUEST_COUNT)) {
-      RequestCountMonitor.updateTodaysRequestNumber(1);
-    } else {
-      logger.info("Feature flag is toggled off for request number counter update");
-    }
+  const isSubmitted: boolean = req.chSession.data[keys.SUBMITTED];
+  if (!isSubmitted) {
+    req.chSession.data[keys.SUBMITTED] = true;
+    await saveSession(req.chSession).then(async () => {
```

Comment:
the session save should probably happen after the client is called. if theres an error with the processor then they wont be able to try again until the session is cleared

Reply:
It depends how likely this is; I'd like to ensure as much as possible that no second click could trigger another processor especially now that CHIPS call is in there.

Reply:
Thats trading a minor inconvenience with potentially a nasty bug and is definitely not worth he trade.